app/server/utils/metrics/metrics.py

```diff
@@ -0,0 +1,201 @@
+        date_filters.append(CreditTransfer.created >= start_date)
+        date_filters.append(CreditTransfer.created <= end_date)
+
+    # Disable cache if any filters are being used, or explicitly requested
+    enable_cache = True
+    if user_filter or date_filters or disable_cache:
+        enable_cache = False
+
+    # Metrics taxonomy, used to determine which metrics to display when which 
+    # metric_type is queried
+    transfer_stats = [
+        transfer_stats.total_distributed, 
+        transfer_stats.total_spent, 
+        transfer_stats.total_exchanged, 
+        transfer_stats.exhausted_balance_count, 
+        transfer_stats.has_transferred_count,
+        transfer_stats.daily_transaction_volume, 
+        transfer_stats.daily_disbursement_volume, 
+        transfer_stats.transfer_use_breakdown
+    ]
+
+    participant_stats = [
+        participant_stats.total_beneficiaries,
+        participant_stats.total_vendors,
+    ]
+    all_stats = transfer_stats + participant_stats
+
+    metric_sets_by_type = {
+        metrics_const.ALL: all_stats,
+        metrics_const.TRANSFER: transfer_stats,
+        metrics_const.PARTICIPANT: participant_stats
+    }
+
+    data = {}
+    for metric in metric_sets_by_type[metric_type]:
+        data[metric.metric_name] = metric.execute_query(user_filters=user_filter, date_filters=date_filters, enable_caching=enable_cache)
+
+    # Legacy and aggregate metrics which don't fit the modular pattern
+    if metric_type in [metrics_const.ALL, metrics_const.PARTICIPANT]:
+        data['total_users'] = data['total_vendors'] + data['total_beneficiaries']
+
+    if metric_type in [metrics_const.ALL, metrics_const.TRANSFER]:
+        try:
+            last_day = data['daily_transaction_volume'][0][1]
+            data['last_day_volume'] = data['daily_transaction_volume'][0][0]
+            data['daily_transaction_volume'] = [
+                {'date': item[1].isoformat(), 'volume': item[0]} for item in data['daily_transaction_volume']
+            ]
+        except IndexError:  # No transactions
+            last_day = datetime.datetime.utcnow()
+            last_day_volume = 0
+            data['has_transferred_count'] = 0
+            data['daily_transaction_volume'] = [{'date': datetime.datetime.utcnow().isoformat(), 'volume': 0}]
+        
+        data['last_day_volume']: {'date': last_day.isoformat(), 'volume': last_day_volume}
+
+        try:
+            data['daily_disbursement_volume'] = [
+                {'date': item[1].isoformat(), 'volume': item[0]} for item in data['daily_disbursement_volume']
+            ]
+        except IndexError:
+            data['daily_disbursement_volume'] = [{'date': datetime.datetime.utcnow().isoformat(), 'volume': 0}]
+
+        try:
+            data['master_wallet_balance'] = cached_funds_available()
+        except:
+            data['master_wallet_balance'] = 0
+
+    return data
+
+
+def cached_funds_available(allowed_cache_age_seconds=60):
```

Comment:
This function is a bit rough, but it's literally just moved over from the old path. Probably a worthwhile todo to rewrite it at some point, but just found it out of scope for here